pkg/querier/worker_service.go

```diff
@@ -64,14 +63,6 @@ func InitWorkerService(
 		serverutil.ResponseJSONMiddleware(),
 	)
 
-	internalRouter := mux.NewRouter()
-	if queryRouterPathPrefix != "" {
-		internalRouter = internalRouter.PathPrefix(queryRouterPathPrefix).Subrouter()
-	}
-	for route, handler := range queryRoutesToHandlers {
-		internalRouter.Path(route).Methods("GET", "POST").Handler(handler)
-	}
-
```

Comment:
this looks fine, but going to need to take a bit more time to dig into this change specifically by running locally in a few different modes, unless you have other ideas on how to test?